File: aplicação/backend/src/main/java/eaproject/output/GetGradeByIdOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;

public class GetGradeByIdOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private int id;

    @JsonProperty
    private String name;

    @JsonProperty
    private int minAge;

    @JsonProperty
    private int maxAge;

    @JsonProperty
    private String gender;

    @JsonProperty
    private Timestamp creationDate;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getMinAge() {
        return minAge;
    }

    public void setMinAge(int minAge) {
        this.minAge = minAge;
    }

    public int getMaxAge() {
        return maxAge;
    }

    public void setMaxAge(int maxAge) {
        this.maxAge = maxAge;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public Timestamp getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Timestamp creationDate) {
        this.creationDate = creationDate;
    }
}

File: aplicação/frontend/ea-project/src/models/output/GetGradeByIdOutput.js
import BaseOutput from "./BaseOutput";

export default class GetGradeByIdOutput extends BaseOutput {
  constructor(
    creationDate = new Date(),
    gender = "",
    maxAge = 0,
    minAge = 0,
    name = "",
    id = 0,
    feedbackMessages = []
  ) {
    super(feedbackMessages);
    this.id = id;
    this.name = name;
    this.minAge = minAge;
    this.maxAge = maxAge;
    this.gender = gender;
    this.creationDate = creationDate;
  }
}
